src/content/docs/guides/upgrade-to-biome-v2.mdx

```diff
@@ -0,0 +1,236 @@
+
+Now Biome has more than 300 rules, and some of those rules *conflict with each other*, causing situations where a project can't fix the violations of the rules, because one fix triggers another rule, and vice versa.
+
+We decided to **take a step back**, remove the option, and introduce it in the future in some other form, maybe with different semantics and configuration. We are aware that this breaking change can cause some inconvenience.
+
+The maintainers of the project are already discussing the topic on Discord and GitHub. You are encouraged to take part in the conversation and help us find a good solution.
+
+As *limited* workaround, you can enable the recommended rules, and enable all the linter domains. However, you can't disable single rules, and the domains `react` and `solid` enable rules that go in conflict with each other:
+
+```json title="biome.json"
+{
+  "linter": {
+    "domains": {
+      "next": "all",
+      "react": "all",
+      "test": "all",
+      "solid": "all",
+      "project": "all"
+    },
+    "rules": {
+      "recommended": true
+    }
+  }
+}
+```
+
+### Syntax `assert` no longer supported
+
+The `assert` syntax, which reached Stage 3, is no longer supported, and it was replaced by the `with` syntax.
+
+All LTS versions of Node.js support the new syntax, as well as all browser engines and serverless runtimes.
+
+```diff mark="with" mark="assert"
+-import {test} from "foo.json" assert { for: "for" }
+-export * from "mod" assert { type: "json" }
++import {test} from "foo.json" with { for: "for" }
++export * from "mod" with { type: "json" }
+```
+
+### The linter works differently
+
+In v1, the linter worked as follows:
+- The recommended rules, by default, emit only diagnostics with error.
+- The non-recommended rules needed to be manually enabled, and the user had to decide the severity.
+
+This way of working was a bit different from other linters (ESLint, clippy, golint, etc.), and it was very limited.
+
+In v2, the linter works like other linters do, which means the following:
+- Every rule is associated with a default severity level, suggested by Biome.
+- The recommended rules can emit diagnostics of different severity.
+- Users can now avail of the default severity of a rule, or choose the severity themselves.
+
+If you relied on recommended rules to always emit an error, the `biome migrate` command will set the severity of those rules to `"error"`.
+
+### `style` rules don't emit errors anymore
+
+We received a lot of great feedback in the past months regarding our recommended rules. We realized that it's challenging to balance a "recommended" set of rules for our users. While some really our defaults, some others thought that they were too noisy.
+
+From v2, all rules that belong to the `style` group won't emit an error unless configured otherwise. The `biome migrate` command will update the configuration so they still emit errors. Make sure that the configuration conforms your standards.
```

Comment:
```suggestion
From v2, all rules that belong to the `style` group won't emit an error unless configured otherwise. The `biome migrate` command will update the configuration so they still emit errors if you had them enabled. Make sure that the configuration conforms to your standards.
```